web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js

```diff
@@ -154,6 +154,48 @@ context('Create Route with Service Discovery Upstream', () => {
     });
   });
 
+  it('should edit this route with Consul Service Discovery upstream', function () {
+    cy.visit('/');
+    cy.contains('Route').click();
+    cy.get(selector.nameSelector).type(data.routeName);
+
+    cy.contains('Search').click();
+    cy.contains(data.routeName).siblings().contains('Configure').click();
+
+    cy.get(selector.name).should('value', data.routeName);
+    cy.contains('Next').click({
+      force: true,
+    });
+
+    cy.contains('Nacos').should('exist');
+
+    // set another service discovery
+    cy.get(selector.discovery_type).click({ force: true });
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('Consul').click();
+    });
+    cy.get(selector.service_name).clear().type(`another.${data.serviceName}`);
+
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.contains(data.submitSuccess).should('be.visible');
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+
+    // check if the changes have been saved
+    cy.get(selector.nameSelector).type(data.routeName);
+    cy.contains('Search').click();
+
+    cy.contains(data.routeName).siblings().contains('Configure').click();
+    // ensure it has already changed to edit page
+    cy.get(selector.name).should('value', data.routeName);
+    cy.contains('Next').click({
+      force: true,
+    });
+    cy.get(selector.service_name).should('value', `another.${data.serviceName}`);
```

Comment:
Also need to check Discovery Type is Consul

Reply:
done